include/epicsRingBytes.h

```diff
@@ -26,32 +35,96 @@ extern "C" {
+/**
+ * @brief Delete the ring buﬀer and free any associated memory
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ */
 epicsShareFunc void epicsShareAPI epicsRingBytesDelete(epicsRingBytesId id);
+/**
+ * @brief Read data out of the ringbuffer
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @param value Up to nbytes get moved from the ring buffer to value 
+ * @param nbytes Get nbytes from ringbuffer
+ * @return The number of bytes actually moved is returned.
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesGet(
     epicsRingBytesId id, char *value,int nbytes);
+/**
+ * @brief Move data to the ringbuffer
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @param value Move nbytes from value to the ring buﬀer if there is enough
+ * free space available to hold them. 
+ * @param nbytes Move nbytes to the ringbuffer
+ * @return The number of bytes actually moved is returned, which will
+ * be zero if insuﬃcient space exists.
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesPut(
     epicsRingBytesId id, char *value,int nbytes);
+/**
+ * @brief Make the ring buﬀer empty
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @note Should only be used when both gets and puts are locked out.
+ */
 epicsShareFunc void epicsShareAPI epicsRingBytesFlush(epicsRingBytesId id);
+/**
+ * @brief Return the number of free bytes in the ring buﬀer
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @return The number of free bytes in the ring buﬀer
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesFreeBytes(epicsRingBytesId id);
+/**
+ * @brief Return the number of bytes currently stored in the ring buﬀer
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @return The number of bytes currently stored in the ring buﬀer
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesUsedBytes(epicsRingBytesId id);
+/**
+ * @brief Return the size of the ring buﬀer
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @return Return the size of the ring buﬀer, i.e., nbytes speciﬁed in
+ * the call to epicsRingBytesCreate().
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesSize(epicsRingBytesId id);
+/**
+ * @brief Test if the ring buﬀer is currently empty.
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @return true, false
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesIsEmpty(epicsRingBytesId id);
+/**
+ * @brief Test if the ring buﬀer is currently full.
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @return true, false
+ */
 epicsShareFunc int  epicsShareAPI epicsRingBytesIsFull(epicsRingBytesId id);
+/**
+ * @brief Get the Highwater mark of the ring buffer
+ * @param id RingbufferID returned by a former epicsRingBytesCreate 
+ * @return Actual Highwater mark
```

Comment:
We probably should elaborate a little what we mean by this. How about: "Returns the highest number of bytes the ring buffer contained since the water mark has been reset. A new ring buffer starts with a water mark of 0."